src/anomalib/engine/engine.py

```diff
@@ -348,11 +473,19 @@ def test(
                 anomalib test --config <config_file_path>
                 ```
         """
+        assert (
+            model or self.model
+        ), "`Engine.test()` requires an `AnomalyModule` when it hasn't been passed in a previous run."
```

Comment:
would it be possible to use exception handling here instead of assert. Bandit complains about this stuff for the security checks.